leaderelection/config.go

```diff
@@ -112,6 +112,18 @@ type ComponentConfig struct {
 	LeaseDuration time.Duration
 	RenewDeadline time.Duration
 	RetryPeriod   time.Duration
+	// LeaseName is a function to customize the lease name given the index i.
+	// If not present, a name in format {Component}.{queue-name}.{i}-of-{Buckets}
+	// will be use.
+	// Autoscaler need to know the Lease names to filter out Leases which are not
+	// used for Autoscaler. Instead of exposing the names from leadelection package,
+	// we let Autoscaler to pass them in.
+	LeaseName func(i uint32) string
```

Comment:
This change breaks Eventing.: 
```
=== FAIL: pkg/reconciler/pingsource TestNew (unknown)
    logger.go:130: 2020-08-28T14:20:00.507Z	DPANIC	zap/sugar.go:210	Ignored key without a value.	{"ignored": "required key PING_IMAGE missing value"}
    logger.go:130: 2020-08-28T14:20:00.509Z	FATAL	pingsource/controller.go:76	Error converting leader election configuration to JSON	{"error": "json: unsupported type: func(uint32) string"}
```

Reply:
Why do you serialize it?

Reply:
we pass configurations to the source adapters by value.